### ISSUE TITLE
Bromite unresponsive due to Multiple fingerprinting mitigations patch

### ISSUE THREAD

**Issue by uazo** [bromite/bromite#2352]
### Preliminary checklist

- [X] I have read the [README](https://github.com/bromite/bromite/blob/master/README.md).
- [X] I have searched the existing issues for my problem. This is a new ticket, NOT a duplicate or related to another open issue.
- [X] I have read the [FAQs](https://github.com/bromite/bromite/blob/master/FAQ.md).
- [X] I have updated Bromite to the latest version. The bug is reproducible on this latest version.
- [X] This is a bug report about the Bromite browser; not the website nor F-Droid nor anything else.

### Can the bug be reproduced with corresponding Chromium version?

No

### Bromite version

105.0.5195.41

### Device architecture

x64

### Android version

9

### Device model

emulator

### Changed flags

none

### Is this bug about the SystemWebView?

No

### Is this bug happening in an incognito tab?

No

### Is this bug caused by the adblocker?

No

### Is this bug a crash?

No

### Describe the bug

Due to incorrect management of the noise factor, the value keeps changing making some pages unresponsive

### Steps to reproduce the bug

Try https://putsreq.herokuapp.com/KOHgthCSW3Ehlo7c6cjl/inspect

### Expected behavior

bromite must not freeze

**Comment by uazo** [comment]
I fix it in this way, which seems more correct to me, but I would like to understand if it happens only to me or it happens to something else

```
diff --git a/third_party/blink/renderer/core/dom/document.cc b/third_party/blink/renderer/core/dom/document.cc
index df76c79e9b..69440cd081 100644
--- a/third_party/blink/renderer/core/dom/document.cc
+++ b/third_party/blink/renderer/core/dom/document.cc
@@ -2234,9 +2234,11 @@ void Document::UpdateStyleAndLayoutTreeForThisDocument() {
 #endif

   if (RuntimeEnabledFeatures::FingerprintingClientRectsNoiseEnabled()) {
-    // Precompute -0.0003% to 0.0003% noise factor for get*ClientRect*() fingerprinting
-    noise_factor_x_ = 1 + (base::RandDouble() - 0.5) * 0.000003;
-    noise_factor_y_ = 1 + (base::RandDouble() - 0.5) * 0.000003;
+    if (noise_factor_x_ == 1 || noise_factor_y_ == 1) {
+      // Precompute -0.0003% to 0.0003% noise factor for get*ClientRect*() fingerprinting
+      noise_factor_x_ = 1 + (base::RandDouble() - 0.5) * 0.000003;
+      noise_factor_y_ = 1 + (base::RandDouble() - 0.5) * 0.000003;
+    }
   } else {
     noise_factor_x_ = 1;
     noise_factor_y_ = 1;
diff --git a/third_party/blink/renderer/core/dom/document.h b/third_party/blink/renderer/core/dom/document.h
index b59a8cf538..5e24b567e4 100644
--- a/third_party/blink/renderer/core/dom/document.h
+++ b/third_party/blink/renderer/core/dom/document.h
@@ -2278,8 +2278,8 @@ class CORE_EXPORT Document : public ContainerNode,

   base::ElapsedTimer start_time_;

-  double noise_factor_x_;
-  double noise_factor_y_;
+  double noise_factor_x_ = 1;
+  double noise_factor_y_ = 1;

   Member<ScriptRunner> script_runner_;
   Member<ScriptRunnerDelayer> script_runner_delayer_;

```

**Comment by csagan5** [comment]
It used to be changed only once per DOM load, but then it was changed as it is now; current behaviour is intentional.

**Comment by uazo** [comment]
> current behaviour is intentional.

but fundamentally incorrect. what do you think can be done? Do you remember why it was changed?
I would not like it to be the cause of #2317 too

**Comment by csagan5** [comment]
> but fundamentally incorrect

Why is updating the noise factor incorrect in `Document::UpdateStyleAndLayoutTreeForThisDocument()`?

> Do you remember why it was changed?

The issues are still on GitHub; I do not recall precisely but probably for the reasons described here: https://github.com/bromite/bromite/wiki/Fingerprinting

**Comment by csagan5** [comment]
Fixed in `105.0.5195.147`.
